File: app/src/main/java/com/phoenix/phoenix/entity/Fornitura.java
package com.phoenix.phoenix.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.http.HttpStatusCode;
import org.springframework.http.ResponseEntity;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@Entity
@Table(name = "fornitura")
public class Fornitura {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne
    private Fornitore fornitore;

    @OneToOne(cascade = CascadeType.ALL)
    private Fattura fattura;

    private Date arrivo;

    private Date scadenza;

    private double prezzo;

    private int quantita;

    private String stato;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "fornitura")
    private List<Merce> merci = new ArrayList<>();

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "fornitura")
    private List<Pellicola> pellicole = new ArrayList<>();
}

File: front/src/app/model/fornitura.ts
import { Fattura } from "./fattura";
import { Fornitore } from "./fornitore";
import { Merce } from "./merce";
import { Pellicola } from "./pellicola";

export interface Fornitura{
    id:number;
    fornitore:Fornitore;
    fattura:Fattura;

    tipo:string;
    arrivo:Date;
    scadenza:Date;
    prezzo:number;
    quantita:number;
    stato:string;
    merci:Array<Merce>;
    pellicole:Array<Pellicola>;
}

export class Fornitura{
    public constructor();
    public constructor(
        fornitore?:Fornitore,
        fattura?:Fattura,
        tipo?:string,
        merci?:Array<Merce>
    ) {}
}